crates/bitwarden-ssh/src/import.rs

```diff
@@ -0,0 +1,235 @@
+use ed25519;
+use pkcs8::{der::Decode, pkcs5, DecodePrivateKey, PrivateKeyInfo, SecretDocument};
+use ssh_key::private::{Ed25519Keypair, RsaKeypair};
+
+use crate::{error::SshKeyImportError, SshKey};
+
+const PKCS1_LABEL: &str = "RSA PRIVATE KEY";
+const PKCS8_UNENCRYPTED_LABEL: &str = "PRIVATE KEY";
+const PKCS8_ENCRYPTED_LABEL: &str = "ENCRYPTED PRIVATE KEY";
+const OPENSSH_LABEL: &str = "OPENSSH PRIVATE KEY";
+
+/// Import a PKCS8 or OpenSSH encoded private key, and returns a decoded SshKey,
+/// with the public key and fingerprint, and the private key in OpenSSH format.
+/// A password can be provided for encrypted keys.
+/// # Returns
+/// - `Error::PasswordRequired` if the key is encrypted and no password is provided
+/// - `Error::WrongPassword` if the password provided is incorrect
+/// - `Error::UnsupportedKeyType` if the key type is not supported
+/// - `Error::ParsingError` if the key is otherwise malformed and cannot be parsed
+pub fn import_key(
+    encoded_key: String,
+    password: Option<String>,
+) -> Result<SshKey, SshKeyImportError> {
+    let label = pem_rfc7468::decode_label(encoded_key.as_bytes())
+        .map_err(|_| SshKeyImportError::ParsingError)?;
+
+    match label {
+        PKCS1_LABEL => Err(SshKeyImportError::UnsupportedKeyType),
+        PKCS8_UNENCRYPTED_LABEL => import_pkcs8_key(encoded_key, None),
+        PKCS8_ENCRYPTED_LABEL => import_pkcs8_key(
+            encoded_key,
+            Some(password.ok_or(SshKeyImportError::PasswordRequired)?),
+        ),
+        OPENSSH_LABEL => import_openssh_key(encoded_key, password),
+        _ => Err(SshKeyImportError::ParsingError),
+    }
+}
+
+fn import_pkcs8_key(
+    encoded_key: String,
+    password: Option<String>,
+) -> Result<SshKey, SshKeyImportError> {
+    let der = if let Some(password) = password {
```

Comment:
I think with the refactors this got lost, but since this is a `SecretDocument` the variable name should be updated to reflect it.

```suggestion
    let doc = if let Some(password) = password {
```